fix: paid seated event attendee privacy

dhairyashiil opened the pull request:
## What does this PR do?

- Fixes #25256 
- Fixes CAL-6782 

## Visual Demo

### Tested for both scenarios
1. when toggle is off
2. when toggle is ON

https://github.com/user-attachments/assets/1491fb31-2d93-465f-b3a9-2d63fb6d8e76






<!-- This is an auto-generated description by cubic. -->
---
## Summary by cubic
Fixes attendee privacy for paid seated events by sending seat-specific notifications after payment and honoring seatsShowAttendees. Addresses #25256 and CAL-6782 to prevent exposing other attendees’ details to each buyer.

- **Bug Fixes**
  - Use sendScheduledSeatsEmailsAndSMS for seated events, emailing only the attendee who just completed payment.
  - Defer confirmation emails for paid seats until payment succeeds and respect workflow settings that disable standard host/attendee emails.
  - Link each booking seat to its paymentId and select the paying attendee when sending emails to avoid misrouted confirmations.

- **Migration**
  - Add paymentId to BookingSeat with a unique index to track seat payments.

<sup>Written for commit e002dee17b791718a2c86342c088a0edfc0bb9c0. Summary will update on new commits.</sup>

<!-- End of auto-generated description by cubic. -->

dhairyashiil committed:
fix: paid seated event attendee privacy

dhairyashiil committed:
implemented all cases

dhairyashiil committed:
revert unnecessary frontend code

dhairyashiil committed:
Merge remote-tracking branch 'upstream/main' into fix/paid-seated-event-attendee-privacy

dhairyashiil committed:
remove unnecessary db call

anikdhabal reviewed (changes requested):
In order to fix this issue, let's follow these steps:

1. Store the paymentId in the bookingSeat table.
2. After a successful payment, query the seat using the passed paymentId to find the attendee/seat, and then send them an email

dhairyashiil committed:
fix: Link payment to booking seat to prevent race condition in email delivery

- Add paymentId field to BookingSeat model for direct payment tracking
- Update createNewSeat to link payment after seat creation
- Update RegularBookingService to link payment for first seat bookings
- Modify handlePaymentSuccess to use paymentId for accurate attendee identification
- Update getBooking to include paymentId in bookingSeat queries
- Add database migration for new paymentId column with unique index

This fixes the race condition where payment webhooks arriving out of order
could cause confirmation emails to be sent to the wrong attendee in paid
seated events. Now each seat directly references its payment for accurate
identification.

dhairyashiil committed:
refactor: Remove redundant index and unused field

- Remove redundant regular index on BookingSeat.paymentId (unique index already provides lookup capability)
- Remove unused phoneNumber field from getBooking query (data never used in CalendarEvent)
- Regenerate migration with optimized schema

This reduces storage overhead and write operations while maintaining performance.

dhairyashiil committed:
Merge remote-tracking branch 'upstream/main' into fix/paid-seated-event-attendee-privacy

anikdhabal commented on packages/emails/email-manager.ts:
```diff
@@ -99,7 +99,8 @@ const _sendScheduledEmailsAndSMS = async (
               },
-              attendee
+              attendee,
+              formattedCalEvent.seatsShowAttendees ?? undefined
```
```suggestion
              formattedCalEvent.seatsShowAttendees ?? false
```

dhairyashiil committed:
refactor: Apply code review feedback

- Fix seatsShowAttendees default from undefined to false to prevent unintended attendee hiding
- Remove unnecessary comments from createNewSeat for cleaner code

dhairyashiil committed:
refactor: Remove unnecessary comments from handlePaymentSuccess

dhairyashiil committed:
refactor: Remove unnecessary comments from RegularBookingService

dhairyashiil committed:
Merge branch 'main' into fix/paid-seated-event-attendee-privacy

devin-ai-integration[bot] committed:
Merge upstream/main into fix/paid-seated-event-attendee-privacy

Co-Authored-By: unknown <>
